src/routes/news/containers/WithCustomNewsProvider.js

```diff
@@ -180,50 +177,28 @@ const withCustomNewsProvider = <
   R: {}
 >(
     refresh: (refreshProps: R, dispatch: Dispatch<StoreActionType>) => void
-  ): ((
-  Component: React.ComponentType<S>
-) => React.ComponentType<PropsType<S, R>>) => {
-  return (
-    Component: React.ComponentType<S>
-  ): React.ComponentType<PropsType<S, R>> => {
-    return class extends React.Component<
-      PropsType<S, R>,
-      {| selectedNewsType: string |}
-    > {
+  ): ((Component: React.ComponentType<S>) => React.ComponentType<PropsType<S, R>>) => {
+  return (Component: React.ComponentType<S>): React.ComponentType<PropsType<S, R>> => {
+    return class extends React.Component<PropsType<S, R>, {| selectedNewsType: NewsType |}> {
       static displayName = wrapDisplayName(Component, 'withCustomNewsProvider');
 
-      state = {
-        selectedNewsType: this.getAvailableNewsType()
-      };
-
-      listRef = null;
+      state = { selectedNewsType: this.getAvailableNewsType() };
 
-      getAvailableNewsType (): string {
+      getAvailableNewsType (): NewsType {
         const { cityModel, selectedNewsType } = this.props.innerProps || {}
         const { tunewsEnabled, pushNotificationsEnabled } = cityModel || {}
         let type = LOCAL
         if (selectedNewsType) {
           type = selectedNewsType
         } else if (tunewsEnabled && !pushNotificationsEnabled) {
           type = TUNEWS
-        } else if (pushNotificationsEnabled && !tunewsEnabled) {
-          type = LOCAL
         }
         return type
       }
 
-      selectNewsType = type => {
-        this.setState(
-          {
-            selectedNewsType: type
-          },
-          this.fetchNews
-        )
-      };
+      selectNewsType = type => { this.setState({ selectedNewsType: type }, this.fetchNews) };
```

Comment:
Still some `;`. You should get errors if you run `yarn run lint` I think

Reply:
Naming is a bit difficult here imo, since this also starts fetching, maybe `selectAndFetchNews`?

Reply:
Yes this is a better one